src/lib/registration/ui/migration_repos_workflow.rb

```diff
@@ -466,6 +470,17 @@ def select_migration_repos
         UI::MigrationReposSelectionDialog.run
       end
 
+      # Select products for migration
+      #
+      # It causes the *-release packages to be installed (see bsc#1086818 for
+      # further details).
+      def select_products
+        selected_migration.map(&:identifier).each do |name|
```

Comment:
Could we somehow reuse the [SwMgmt.select_addon_products](https://github.com/yast/yast-registration/blob/4866c70e53c485a45711b53eb9f0ae7e5e1dc497/src/lib/registration/sw_mgmt.rb#L512) here?

Additionally it also selects the default patterns which we should not do in upgrade so we would need a new parameter.

But IMHO it works a bit better because it does not blindly expect the "product name" == "migration identifier", but uses the available product from that repository. Which means it should better handle the inconsistencies between the SCC identifier and the libzypp product. Although that's rather a corner case, normally they should match...

But I do not know if during the migration the used `Registration::Storage::Cache.instance.addon_services` value is set. Please check.

Reply:
That was my first try, but during migration `addon_services` is always empty. But I will try to find out why `addon_services` is empty.